src/Illuminate/Support/Collection.php

```diff
@@ -315,10 +315,12 @@ public function filter(callable $callback = null)
      * @param  callable  $callback
      * @return mixed
      */
-    public function when($value, callable $callback)
+    public function when($value, callable $callback, callable $default = null)
     {
         if ($value) {
             return $callback($this);
+        } elseif ($default) {
+            return $default($this);
```

Comment:
Gonna leave that as is. Thats what we currently use for the eloquent QB and it works just fine for now